sql/analyzer/resolve_tables.go

```diff
@@ -268,7 +268,11 @@ func reresolveTables(ctx *sql.Context, a *Analyzer, node sql.Node, scope *Scope,
 			if n.Database != nil {
 				db = n.Database.Name()
 			}
-			to, err = resolveTable(ctx, plan.NewUnresolvedTable(n.Name(), db), a)
+			var asof sql.Expression
+			if n.AsOf != nil {
+				asof = expression.NewLiteral(n.AsOf, nil)
```

Comment:
Why is this conversion step necessary? Why not just `resolveTable(plan.NewUnresolvedTableAsOf(n.name, db, n.AsOf)`?